File: classifier/data/raw/ai/samples/project_euler_java/Problem026.java
// Project Euler 26: d < 1000 whose reciprocal has the longest recurring cycle.
public class Problem026 {
    private static int cycleLength(int denominator) {
        int[] firstSeen = new int[denominator];
        java.util.Arrays.fill(firstSeen, -1);
        int remainder = 1 % denominator;
        int position = 0;
        while (remainder != 0 && firstSeen[remainder] == -1) {
            firstSeen[remainder] = position++;
            remainder = (remainder * 10) % denominator;
        }
        return remainder == 0 ? 0 : position - firstSeen[remainder];
    }

    public static void main(String[] args) {
        int bestDenominator = 0;
        int bestLength = 0;
        for (int d = 2; d < 1_000; d++) {
            int length = cycleLength(d);
            if (length > bestLength) {
                bestLength = length;
                bestDenominator = d;
            }
        }
        System.out.println(bestDenominator);
    }
}

File: classifier/data/raw/ai/samples/project_euler_zig/problem_026.zig
// Project Euler 026: denominator below 1,000 with the longest recurring cycle.
extern "c" fn write(fd: c_int, buf: [*]const u8, count: usize) isize;

pub fn main() !void {
    var best_denominator: usize = 0;
    var longest: i32 = 0;
    var d: usize = 2;
    while (d < 1_000) : (d += 1) {
        var first_seen: [1_000]i32 = undefined;
        for (&first_seen) |*position| position.* = -1;
        var remainder: usize = 1 % d;
        var position: i32 = 0;
        while (remainder != 0 and first_seen[remainder] == -1) {
            first_seen[remainder] = position;
            remainder = (remainder * 10) % d;
            position += 1;
        }
        if (remainder != 0) {
            const cycle = position - first_seen[remainder];
            if (cycle > longest) {
                longest = cycle;
                best_denominator = d;
            }
        }
    }

    var buffer: [32]u8 = undefined;
    var pos = buffer.len - 1;
    buffer[pos] = '\n';
    var value = best_denominator;
    while (value > 0) : (value /= 10) {
        pos -= 1;
        buffer[pos] = @intCast('0' + value % 10);
    }
    _ = write(1, buffer[pos..].ptr, buffer.len - pos);
}
